### PR TITLE
Changes to helm-charts for pushing auditlogs to postgres

### PR DESCRIPTION
**Pull request by vivekhiwarkar**
### What does this PR change?

- Changes to helm for pushing auditlogs to postgres by default 

### Does the PR depend on any other PRs or Issues? If yes, please list them.

- No

### Checklist

I confirm, that I have...

- [X] Read and followed the contributing guide in `CONTRIBUTING.md`
- [ ] Updated [documentation on the Paralus docs site](https://github.com/paralus/website/blob/main/docs) (if applicable)
- [ ] Updated `CHANGELOG.md`
- [x] Ran [`helm-docs`](https://github.com/norwoodj/helm-docs) to update docs for chart (if values.yaml file was changed)

### CODE REVIEW

File: charts/ztka/templates/_helpers.tpl [hunk 1]
```diff
@@ -69,10 +82,12 @@ http://{{.Release.Name}}-kratos-admin
 Get Elasticsearch Address.
 */}}
 {{- define "ztka.esAddr" -}}
-  {{- if .Values.deploy.elasticsearch.enable -}}
+  {{- if eq .Values.auditLogs.storage "elasticsearch" -}}
```
Comment (meain) [inline charts/ztka/templates/_helpers.tpl:85]: Can we use `ztka.auditlogStorage` here instead?

File: charts/ztka/values.yaml [hunk 2]
```diff
@@ -232,6 +237,11 @@ deploy:
     indexPrefix: "ralog"
     # TODO: can we somehow template it in filebeat.yml?
 
+  fluentbit:
+    # -- Filebeat is used to collect audit logs into the system. You
```
Comment (akshay196) [inline charts/ztka/values.yaml:241]:
```suggestion
    # -- Fluentbit is used to collect audit logs into the system. You
```

File: charts/ztka/templates/configmap-paralus.yaml [hunk 3]
```diff
@@ -8,6 +8,7 @@ metadata:
 data:
   DEV: "false"
   AUDIT_LOG_FILE: "/audit-logs/audit.log"
+  AUDIT_LOG_STORAGE: {{ include "ztka.auditlogStorage" . | quote }}
```
Comment (akshay196) [inline charts/ztka/templates/configmap-paralus.yaml:11]: Why is the use of "AUDIT_LOG_STORAGE" variable in Paralus?

File: charts/ztka/templates/_helpers.tpl [hunk 4]
```diff
@@ -35,6 +35,19 @@ Create the name of the service account to use
 {{- end }}
 {{- end }}
 
+{{/*
+Validate storage options
+*/}}
+{{- define "ztka.auditlogStorage" -}}
+  {{- if eq .Val .Values.auditLogs.storage "postgres" -}}
```
Comment (akshay196) [inline charts/ztka/templates/_helpers.tpl:42]:
```suggestion
  {{- if eq .Values.auditLogs.storage "postgres" -}}
```

File: charts/ztka/templates/configmap-fluentbit.yaml [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+
+    [OUTPUT]
+        Name pgsql
+        Match *
+        Host {{ include "ztka.dbAddr" . | }}
+        Port 5432
+        User {{ include "ztka.dbUser" . | }}
+        Password {{ include "ztka.dbPassword" . | }}
+        Database {{ include "ztka.dbName" . | }}
+        Table paralus
```
Comment (akshay196) [inline charts/ztka/templates/configmap-fluentbit.yaml:89]: Use some different table name, like `audit_logs`?

**Review by akshay196 — CHANGES_REQUESTED** [review]
Few minor changes/question.
I will give it a try locally, Will share more comments later.

File: charts/ztka/templates/configmap-paralus.yaml (hunk 3 above)
Comment (vivekhiwarkar) [reply charts/ztka/templates/configmap-paralus.yaml:11]: Based on this variable (postgres/elasticsearch) we will push auditLogs to respective db

File: charts/ztka/templates/configmap-fluentbit.yaml (hunk 5 above)
Comment (vivekhiwarkar) [reply charts/ztka/templates/configmap-fluentbit.yaml:89]: yes, i was thinking the same will change this in the next commit

File: charts/ztka/templates/configmap-paralus.yaml (hunk 3 above)
Comment (akshay196) [reply charts/ztka/templates/configmap-paralus.yaml:11]: Got it.

**Review by akshay196 — COMMENTED** [review]
I am getting following entries in table which we should not add:
```
```

**Comment by akshay196** [comment]
We can parse the timestamp out of record using another json formatted parser. Example:
```
$ fluent-bit -c fluent-bit.conf -R custom_parsers.conf 
Fluent Bit v1.8.12
* Copyright (C) 2019-2021 The Fluent Bit Authors
* Copyright (C) 2015-2018 Treasure Data
* Fluent Bit is a CNCF sub-project under the umbrella of Fluentd
* https://fluentbit.io

[2022/10/31 16:26:44] [ info] [engine] started (pid=352178)
[2022/10/31 16:26:44] [ info] [storage] version=1.1.5, initializing...
[2022/10/31 16:26:44] [ info] [storage] in-memory
[2022/10/31 16:26:44] [ info] [storage] normal synchronization mode, checksum disabled, max_chunks_up=128
[2022/10/31 16:26:44] [ info] [cmetrics] version=0.2.2
[2022/10/31 16:26:44] [ info] [sp] stream processor started
[2022/10/31 16:26:44] [ info] [input:tail:tail.0] inotify_fs_add(): inode=84035909 watch_fd=1 name=tail.log
[0] system: [1666245900.238000000, {"version"=>"1.0", "category"=>"AUDIT", "origin"=>"core", "actor"=>{"type"=>"USER", "account"=>{"username"=>"admin@paralus.local"}, "groups"=>["Organization Admins", "All Local Users"]}, "client"=>{"type"=>"BROWSER", "ip"=>"10.244.0.6:43164", "user_agent"=>"Mozilla/5.0 (X11; Linux x86_64; rv:105.0) Gecko/20100101 Firefox/105.0", "host"=>"console.paralus.local"}, "detail"=>{"message"=>"User test@example.com created", "meta"=>{"username"=>"test@example.com"}}, "type"=>"user.create.success", "portal"=>"OPS", "project"=>""}]
```

Parsers:
```
[PARSER]
    Name container-log
    Format regex
    Regex ^(?<time>[^ ]+) (?<stream>stdout) (?<logtag>[^ ]*) (?<log>.*)$
[PARSER]
    Name json
    Format json
    Time_Key timestamp
    Time_Format %Y-%m-%dT%H:%M:%S.%LZ
```
Fluent bit config:
```
[OUTPUT]
    Name stdout
    Match *
[INPUT]
    Name tail
    Tag system
    Path tail.log
    Refresh_Interval 5
    # DB /var/log/fluentbit.db
    DB.Sync Normal
    Mem_Buf_Limit 5M
    Skip_Long_Lines On
    Read_from_Head True
    Parser container-log
[FILTER]
    Name parser
    Match *
    Key_Name log
    Parser json
```
Example tail log used:
```
2022-10-20T08:21:41.97727022Z stdout F {"timestamp":"2022-10-20T06:05:00.238Z","version":"1.0","category":"AUDIT","origin":"core","actor":{"type":"USER","account":{"username":"admin@paralus.local"},"groups":["Organization Admins","All Local Users"]},"client":{"type":"BROWSER","ip":"10.244.0.6:43164","user_agent":"Mozilla/5.0 (X11; Linux x86_64; rv:105.0) Gecko/20100101 Firefox/105.0","host":"console.paralus.local"},"detail":{"message":"User test@example.com created","meta":{"username":"test@example.com"}},"type":"user.create.success","portal":"OPS","project":""}
```

**Comment by akshay196** [comment]
@vivekhiwarkar Why do we need `kubernetes` and `record_modifier` filters?

**Comment by akshay196** [comment]
Moved to new PR: https://github.com/paralus/helm-charts/pull/67

**Pull request closed without merging**